scripts/west_commands/runners/nrf_common.py

```diff
@@ -307,6 +315,19 @@ def program_hex(self):
                 self.build_conf.getboolean('CONFIG_SOC_NRF54H20_ENGB_CPURAD') or
                 self.build_conf.getboolean('CONFIG_SOC_NRF9280_CPURAD')
             )
+            generated_uicr = self.build_conf.getboolean('CONFIG_NRF_REGTOOL_GENERATE_UICR')
+
+            if cpuapp:
+                core = 'NRFDL_DEVICE_CORE_APPLICATION'
+            elif cpurad:
+                core = 'NRFDL_DEVICE_CORE_NETWORK'
+
+            if generated_uicr and not self.hex_get_uicrs().get(core):
+                raise RuntimeError(
+                    f"Expected a UICR to be contained in: {self.hex_}\n"
+                    "Please ensure that the correct version of nrf-regtool is "
+                    "installed, then run 'west build --cmake' to try again."
```

Comment:
why is `--cmake` required? maybe:
```suggestion
                    "installed, then rebuild and try again."
```

Reply:
You need to re-run CMake to generate the UICR after installing regtool; it will not re-run automatically.